Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,592 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/NSDecimal.h>
+#import <StubReturn.h>
+#include <algorithm>
+
+#define EXPONENT_MAX 127
+#define EXPONENT_MIN -128
+
+#define MANTISSA(decimal) decimal->_mantissa
+
+#define LENGTH(decimal) decimal->_length
+
+#define EXPONENT(decimal) decimal->_exponent
+
+#define NEGATIVE_FLAG(decimal) decimal->_isNegative
```

Comment:
>decimal [](start = 31, length = 7)

If you put the macro argument in parentheses here, you wouldn't have to use double parentheses when calling the macro, as you do later on.